Autopilot/Path Manager/MainBatterySensor.c

```diff
@@ -6,40 +6,34 @@
  */
 #include "main.h"
 #include "../Common/Common.h"
-#include "voltageSensor.h"
+#include "MainBatterySensor.h"
 
 
-char percent = 0;
-int currentVoltage = 0;
+int batteryVoltage = 0;
 
 void __attribute__((interrupt, no_auto_psv)) _ADC2Interrupt(void)
 {
 
-    currentVoltage = ADC2BUF0;
+    batteryVoltage = ADC2BUF0;
+        
     IFS1bits.AD2IF = 0;		// Clear the ADC Interrupt Flag
 
 
 }
 
-void initBatterySensor(){
+void initMainBatterySensor(){
     //AN12 is the pin to get the battery information
     TRISBbits.TRISB12 = 1;
-    initVoltageADC();
+    initMainBatteryADC();
 
 }
 
-//float timeRemaining(){
-//    float dP = (currentPercent - lastPercent)/vTimeInterval; //Rate of change (Percent/second)
-//    float time = currentPercent/dP;
-//    return time;
-//}
-
-int getBatteryLevel(){
-    percent = currentVoltage;
-    return percent;
+int getMainBatteryLevel(){
+//return raw voltage value
+    return batteryVoltage;
 }
 
-void initVoltageADC(){
+void initMainBatteryADC(){
```

Comment:
I think you got the right idea and I guess it's just my bad code, but I've
been making all my functions public on the entire Picpilot.

It might be something we want to gradually fix.

On Jul 22, 2016 2:55 PM, "Serge Babayan" <notifications@github.com> wrote:

> In Autopilot/Path Manager/MainBatterySensor.c
> <https://github.com/UWARG/PICpilot/pull/56#discussion_r71928296>:
>
> >  }
> >
> > -void initVoltageADC(){
> > +void initMainBatteryADC(){
>
> we're not using this method anywhere, thus theres no need for it to be
> available publically. Add a static property to this method, so that it can
> only be used within the same file, and remove it from the MainBattery.h
> file.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWARG/PICpilot/pull/56/files/96c0c8c9d151c4d0ad8ec1bc44304eaadc568dc2#r71928296>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC4dF3QSqC0uKV7bI4sM0deLuBEKwYX7ks5qYRI8gaJpZM4JRHQr>
> .
>